### PR TITLE
mk/aosp_optee.mk: allow building ta with specified target architecture

### PR DESCRIPTION
**Pull request by taddk**
Introduce $(local_optee_ta_target) to specify the target architecture of the built TA in aosp build environment.
Since some ta might prefer better performance with ta_arm64, and some prefer smaller code size with ta_arm32.

### CODE REVIEW

File: mk/aosp_optee.mk [hunk 1]
```diff
@@ -115,6 +115,10 @@ LOCAL_MODULE_TAGS := optional
 LOCAL_REQUIRED_MODULES := $(local_module_deps)
 
 TA_TMP_DIR := $(subst /,_,$(LOCAL_PATH))
+ifneq ($(filter-out $(OPTEE_TA_TARGETS),$(local_optee_ta_target)),)
+TA_DEV_KIT_DIR := $(TA_DEV_KIT_DIR:$(OPTEE_TA_TARGETS)=$(local_optee_ta_target))
```
Comment (jenswi-linaro) [inline mk/aosp_optee.mk:119]: Why not assign the desired value the first time around line 58?
What is `$(TA_DEV_KIT_DIR:$(OPTEE_TA_TARGETS)=$(local_optee_ta_target))` supposed to do?

File: mk/aosp_optee.mk [hunk 2]
```diff
@@ -115,6 +115,10 @@ LOCAL_MODULE_TAGS := optional
...
+TA_TMP_DIR := $(addsuffix _$(local_optee_ta_target), $(TA_TMP_DIR))
```
Comment (jenswi-linaro) [inline mk/aosp_optee.mk:120]: Why is this needed?

File: mk/aosp_optee.mk (hunk 1 above)
Comment (taddk) [reply mk/aosp_optee.mk:119]: Indeed, it is possible to move this to line 58.
We do the substitution to replace the default configuration set by `$(OPTEE_TA_TARGETS)` to `$(local_optee_ta_target)` specified by module Android.mk. I think we can move this to line 58 like,
```
ifneq ($(local_optee_ta_target),)
TA_DEV_KIT_DIR := $(ABS_OPTEE_OS_OUT_DIR)/export-$(local_optee_ta_target)
else
TA_DEV_KIT_DIR := $(ABS_OPTEE_OS_OUT_DIR)/export-${OPTEE_TA_TARGETS}
endif
```

File: mk/aosp_optee.mk (hunk 2 above)
Comment (taddk) [reply mk/aosp_optee.mk:120]: For a common static library that could be linked by several TAs (in different architectures), we would put the intermediate object and library files in different architecture to different folder. Also, it helps the TA in specified architecture (different from the default one) to find the correct folder of the static library in the same architecture.

File: mk/aosp_optee.mk (hunk 1 above)
Comment (jenswi-linaro) [reply mk/aosp_optee.mk:119]: Sounds good

File: mk/aosp_optee.mk (hunk 2 above)
Comment (jenswi-linaro) [reply mk/aosp_optee.mk:120]: OK

Comment (taddk) [reply mk/aosp_optee.mk:123]: Would that be acceptable if we also change the original path by adding the `"_ta_{target_arch}"` to the `$(TA_TMP_DIR)` no matter `$(local_optee_ta_target)` holds a value? like,

```
ifneq ($(local_optee_ta_target),)
TA_TMP_DIR := $(addsuffix _$(local_optee_ta_target), $(subst /,_,$(LOCAL_PATH)))
else
TA_TMP_DIR := $(addsuffix _$(OPTEE_TA_TARGETS), $(subst /,_,$(LOCAL_PATH)))
endif
```

Comment (jenswi-linaro) [reply mk/aosp_optee.mk:123]: Sure. In fact that should allow using a helper variable instead of `local_optee_ta_target` and `OPTEE_TA_TARGETS`.

Comment (taddk) [reply mk/aosp_optee.mk]: Good idea. Updated.

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by taddk** [comment]
Tag applied.

Sorry for late reply. if anything need to be improved or changed, just let me know.
